### PR TITLE
AppVeyor MSYS2 Environment

### PR DESCRIPTION
**Pull request by RobertBColton**
This is just a minor tweak to our AppVeyor build script that makes it less error prone. I don't like having the MSYS2 environment hardcoded like we did and found a way around it by launching our make/build from the `msys2_shell.cmd` launcher batch script. This is arguably more "technically" correct because it's less likely to break if MSYS2 or AppVeyor deploys a change that requires that environment be updated. 

Basically, this change amortizes the bus factor we have on this project right now 😋

### CODE REVIEW

**Comment by RobertBColton** [comment]
@JoshDreamland really simple change here that I need a proper review on, thanks!
